backend/src/main/java/ai/verta/modeldb/config/Config.java

```diff
@@ -81,18 +82,31 @@ public FutureJdbi getJdbi() {
     if (this.jdbi == null) {
       // Initialize HikariCP and jdbi
       final var databaseConfig = config.database;
-      final var hikariDataSource = new HikariDataSource();
-      final var dbUrl = CommonHibernateUtil.buildConnectionString(databaseConfig.RdbConfiguration);
-      hikariDataSource.setJdbcUrl(dbUrl);
-      hikariDataSource.setUsername(databaseConfig.RdbConfiguration.RdbUsername);
-      hikariDataSource.setPassword(databaseConfig.RdbConfiguration.RdbPassword);
-
-      hikariDataSource.setMaximumPoolSize(10); // TODO: configure with a valid pool size
+      return initializeJdbi(databaseConfig);
+    }
+    return this.jdbi;
+  }
 
-      final Jdbi jdbi = Jdbi.create(hikariDataSource);
-      final Executor dbExecutor = FutureGrpc.initializeExecutor(databaseConfig.threadCount);
-      this.jdbi = new FutureJdbi(jdbi, dbExecutor);
+  public FutureJdbi getTestJdbi() {
+    if (this.jdbi == null) {
+      // Initialize HikariCP and jdbi
+      final var databaseConfig = config.test.database;
+      return initializeJdbi(databaseConfig);
     }
     return this.jdbi;
   }
+
+  private FutureJdbi initializeJdbi(DatabaseConfig databaseConfig) {
+    final var hikariDataSource = new HikariDataSource();
+    final var dbUrl = CommonHibernateUtil.buildConnectionString(databaseConfig.RdbConfiguration);
+    hikariDataSource.setJdbcUrl(dbUrl);
+    hikariDataSource.setUsername(databaseConfig.RdbConfiguration.RdbUsername);
+    hikariDataSource.setPassword(databaseConfig.RdbConfiguration.RdbPassword);
+
+    hikariDataSource.setMaximumPoolSize(10); // TODO: configure with a valid pool size
+
+    final Jdbi jdbi = Jdbi.create(hikariDataSource);
+    final Executor dbExecutor = FutureGrpc.initializeExecutor(databaseConfig.threadCount);
+    return new FutureJdbi(jdbi, dbExecutor);
```

Comment:
We're no longer saving `this.jdbi` anywhere. We create a new instance every single time the `get` methods are called.

Reply:
We are checking `if (this.jdbi == null) {` before calling `initializeJdbi` method so if it was already initialized then we are return existing one.

Reply:
Yes. fixed this. it is a bug.